### PR TITLE
Bogavril/nuget

### PR DESCRIPTION
**Pull request by bgavrilMS**
Adds a readme to the MSAL package. 

Manually tested that it works as per https://devblogs.microsoft.com/nuget/add-a-readme-to-your-nuget-package/

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,24 +7,6 @@ The MSAL library for .NET is part of the [Microsoft identity platform for develo
...
-
-**Microsoft.Identity.Client.Broker**  
-[![NuGet](https://img.shields.io/nuget/v/Microsoft.Identity.Client.Broker.svg?style=flat-square&label=nuget&colorB=00b200)](https://www.nuget.org/packages/Microsoft.Identity.Client.Broker/)
-
-**Microsoft.Identity.Client.Extensions.Msal**  
-[![NuGet](https://img.shields.io/nuget/v/Microsoft.Identity.Client.Extensions.Msal.svg?style=flat-square&label=nuget&colorB=00b200)](https://www.nuget.org/packages/Microsoft.Identity.Client.Extensions.Msal/)  
-
-## Build Status
-
-[![Build Status](https://identitydivision.visualstudio.com/IDDP/_apis/build/status/CI/DotNet/.NET%20MSAL%20CI%20(YAML)?branchName=main)](https://identitydivision.visualstudio.com/IDDP/_build/latest?definitionId=922&branchName=main)
```
Comment (bgavrilMS) [inline README.md:26]: pipeline never ran :)

File: README.md [hunk 2]
```diff
@@ -45,10 +27,6 @@ MSAL.NET became Generally Available with MSAL.NET 3.0.8. Since MSAL.NET moved to
 
 - Guides, tutorials, and detailed walkthroughs are available [on Microsoft Learn](https://learn.microsoft.com/entra/msal/dotnet/getting-started/choosing-msal-dotnet).
 - API documentation is available [on Microsoft Learn](https://learn.microsoft.com/dotnet/api/microsoft.identity.client)
-- A number of quickstarts are available for:
```
Comment (bgavrilMS) [inline README.md:48]: too many words

File: src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs [hunk 3]
```diff
@@ -316,6 +316,11 @@ public ConfidentialClientApplicationBuilder WithCacheSynchronization(bool enable
             return this;
         }
 
+        /// <summary>
+        ///
```
Comment (trwalke) [inline src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs:320]: Missing comment?

File: README.md (hunk 1 above)
Comment (trwalke) [reply README.md:26]: Seems like it is pointing to the wrong pipeline

File: README.md [hunk 4]
```diff
@@ -7,24 +7,6 @@ The MSAL library for .NET is part of the [Microsoft identity platform for develo
...
 | ------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------ | ------------------------------------------------------------------------------------------------------------------ | ----------------------------------------------- | ------------------------------------------------------------------------------------------------------- |
 
-## NuGet packages
-
-**Microsoft.Identity.Client**  
-[![NuGet](https://img.shields.io/nuget/v/Microsoft.Identity.Client.svg?style=flat-square&label=nuget&colorB=00b200)](https://www.nuget.org/packages/Microsoft.Identity.Client/)
```
Comment (gladjohn) [inline README.md:13]: this is one way after every release I verify if the NuGets have been updated for all packages :(

File: src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj [hunk 5]
```diff
@@ -114,6 +114,7 @@
     <Compile Remove="$(PathToMsalSources)\Platforms\**\*.*;$(PathToMsalSources)\Resources\*.cs" />
     <Compile Remove="$(PathToMsalSources)\PlatformsCommon\PlatformNotSupported\ApiConfig\SystemWebViewOptions.cs" />
     <EmbeddedResource Include="$(PathToMsalSources)\Properties\Microsoft.Identity.Client.rd.xml" />
+    <None Include="$(PathToMsalSources)\..\..\..\README.md" Pack="true" PackagePath="\"/>
```
Comment (gladjohn) [inline src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj:117]: will this be packed into the NuGet? all packages?

**Review by gladjohn — APPROVED** [review]

File: src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs (hunk 3 above)
Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs:320]: No, not missing. Method is obsolete.

File: src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj (hunk 5 above)
Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj:117]: No, this is included in `MSAL.csproj` so it only affects MSAL.

File: README.md (hunk 1 above)
Comment (bgavrilMS) [reply README.md:26]: The point here is to use this readme.md for nuget package description. Build status doesn't make sense.

File: README.md (hunk 4 above)
Comment (bgavrilMS) [reply README.md:13]: The point here is to use this readme.md for nuget package description. We can add these labels to a separate .md file if needed.

File: src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs (hunk 3 above)
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs:320]:
```suggestion
        /// Call <see cref="WithOidcAuthority(string)"/> instead.

```

File: README.md (hunk 4 above)
Comment (pmaytak) [reply README.md:13]: I usually check the nuget.org itself, since it's closer to the source of truth rather than looking through the labels.

**Review by pmaytak — APPROVED** [review]

**Pull request merged**